include/nuttx/sched_note.h

```diff
@@ -105,16 +105,38 @@
   memset((s), 0, sizeof(struct note_filter_irq_s))
 #endif
 
-/* Note dump module tag definitions */
-
 #ifdef CONFIG_SCHED_INSTRUMENTATION_DUMP
-#  define NOTE_MODULE(a, b, c, d)  \
-  ((uint32_t)((a) & 0xff)        | \
-  ((uint32_t)((b) & 0xff) << 8)  | \
-  ((uint32_t)((c) & 0xff) << 16) | \
-  ((uint32_t)((d) & 0xff) << 24))
+#  define SCHED_NOTE_LABEL__(x, y) x ## y
+#  define SCHED_NOTE_LABEL_(x, y) SCHED_NOTE_LABEL__(x, y)
+#  define SCHED_NOTE_LABEL \
+          SCHED_NOTE_LABEL_(sched_note_here, __LINE__)
+#  define SCHED_NOTE_IP \
+          ({SCHED_NOTE_LABEL: (uintptr_t)&&SCHED_NOTE_LABEL;})
```

Comment:
This is a good improvement, but still I have a question why 
```
#  define SCHED_NOTE_STRING(buf) \
          sched_note_string(__LINE__, buf)
```
should not work? I mean why do we need `&&`?

Reply:
I mean that `&&label` itself is a GNU extension.

Reply:
I think I know why. Because we need a unique value

Reply:
Let me think a bit. Because from compiler compatibility point of view both `__label__` or reworked mens the same if `&&` is used.

Reply:
https://gcc.gnu.org/onlinedocs/gcc/Labels-as-Values.html